### DISCUSSION TITLE
[User Opinion]: Warning - GitHub Classroom is no longer a viable solution for instructors to distribute assignments

### DISCUSSION THREAD

**Discussion by firasm** [community/community#137078] [GitHub Education]
### Hi everyone,

Hello - As the title says, just a quick note to fellow instructors: at the moment GH classroom is just not stable/viable solution for instructors wanting to distribute assignments via GitHub repos in their classes this fall term. 

I know the GH Classroom team is working hard and two betas have come and gone this summer. But it’s very unfortunate the product is now in a state where A) everyone’s workflow needed to change, B) there were significant feature regressions, and C) the student and instructor experience is significantly worse. I wish I could pay to get some support for this product and talk to a real human!

Overall, I wish I had never asked for the ability to update my starter code after an assignment had been released. That one feature request (that we all wanted) seems to have led to a cascade of decisions that’s just broken everything 😭. 

Oh well - these things happen in software sometimes (especially on the eve of a new term!). 

The question is, what we do next. What are people doing instead?

- Template repos?
- Forks?
- Ask students to just create new repos in an org and manually move files over?
- Script the creation/provision of repos to students in an org?

**Comment by aaryno** [comment]
This is really aggravating. Every time I start a new semester it takes about 2 hours just to do the copy/pasta of migrating all the assignments. Last semester I made the mistake of using the beta auto-migration and had to manually delete/recreate everything after toggling template repos to private and then back to public. It took about 5 hours. This semester I am finding that when I create an assignment my template repo is forked and copied to my new org, breaking the link to the raw source templates which I'm still editing -- so I guess I'll have to go back and re-create these assignments? This was such a promising method for managing my class while giving students exposure to git. Now I'm regretting everything.

**Comment by aaryno** [comment]
Just getting started with this newly changed flow so we will see how it goes. This is the setup I have:
1) An organization with all my template repos. All template repos are public. I call these my reference repos.
2) Every semester I create a new org for my class.
3) I create a Google Classroom for the new org and class
4) I create new assignments in the Google Classroom I just created for my new semester. One at a time. 
5) For each assignment:
   - I select the template from my reference repo.
   - Set repository as private so students have privacy
6) Copy the Google Classroom link for the assignment into a master doc in the syllabus.
7) For grading I browse the repositories for Pull Requests after the due date. The PRs contain the student's assignment work. There is some limited CI that does checks for their commits to ensure that they have the right names and types of files. That's how our class assignment submission works. I'm not using autograding or due dates within the Google Classroom. 

Previously, the new assignments were generated from the templates directly. As I was building out the class I could update the templates and have those immediately reflected in any new assignment checkouts.

Now, Google classroom has copied the reference repos to my new org at the point in time when I created the assignment. If I want to make changes to the reference repo it looks like I will need to update the new org's generated template and then push those back upstream to the reference repo. This is going to be a PITA.

**Comment by ValentinoSpirito** [comment]
Hey Firas,

I completely understand the frustration, especially with the term starting soon. It sounds like the recent changes in GitHub Classroom have made it tough for instructors to keep workflows smooth and assignments manageable.

For alternatives, here are a few suggestions you might consider:

Template Repositories: This is a good way to create a base that students can duplicate. It’s easier for them to work with independently without impacting the original.
Forking: Forks could work, but they do have some limitations, especially when it comes to managing updates to starter code.
Manual Creation with an Organization: Asking students to create new repos in an org manually or scripting the creation process can give you more control, but can also be cumbersome depending on the class size.
Scripting Automation: A script for provisioning repos might strike a balance—automating the setup while keeping things organized under your org.
Hopefully, one of these options can help until GitHub Classroom is back to a more stable state.

Let me know if you need any more help or further ideas!

Best,
Valentino

**Reply by firasm** [reply]
Thanks! I ended up going with templates and let go of having all the repos in the term org 🤷‍♂️
